packages/@aws-cdk/aws-codestarnotifications/lib/notification-rule.ts

```diff
@@ -0,0 +1,237 @@
+   *
+   * @default DetailType.FULL
+   */
+  readonly detailType?: DetailType;
+}
+
+/**
+ * Standard set of options for `notifyOn` codestar notification handler on construct
+ */
+export interface NotifyOnEventOptions extends NotifyOptions {
+  /**
+   * A list of event types associated with this notification rule.
+   * For a complete list of event types and IDs, see Notification concepts in the Developer Tools Console User Guide.
+   * @see https://docs.aws.amazon.com/dtconsole/latest/userguide/concepts.html#concepts-api
+   *
+   * You must specify this property (either via props or via `addEvents`)
+   *
+   * @default - No events.
+   */
+  readonly events?: string[];
+}
+
+/**
+ * Properties for a new notification rule
+ */
+export interface NotificationRuleProps extends NotifyOnEventOptions {
+  /**
+   * The Amazon Resource Name (ARN) of the resource to associate with the notification rule.
+   * Currently, Supported sources include pipelines in AWS CodePipeline and build projects in AWS CodeBuild in this L2 constructor.
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-codestarnotifications-notificationrule.html#cfn-codestarnotifications-notificationrule-resource
+   */
+  readonly source: INotificationRuleSource;
+}
+
+/**
+ * Represents a notification rule
+ */
+export interface INotificationRule extends IResource {
+
+  /**
+   * The ARN of the notification rule (i.e. arn:aws:codestar-notifications:::notificationrule/01234abcde)
+   *
+   * @attribute
+   */
+  readonly notificationRuleArn: string;
+
+  /**
+   * Adds target to notification rule
+   *
+   * @param target The SNS topic or AWS Chatbot Slack target
+   * @returns boolean - return true if it had any effect
+   */
+  addTarget(target?: INotificationRuleTarget): boolean;
+}
+
+/**
+ * A new notification rule
+ *
+ * @resource AWS::CodeStarNotifications::NotificationRule
+ */
+export class NotificationRule extends Resource implements INotificationRule {
+  /**
+   * Import an existing notification rule provided an ARN
+   * @param scope The parent creating construct
+   * @param id The construct's name
+   * @param notificationRuleArn Notification rule ARN (i.e. arn:aws:codestar-notifications:::notificationrule/01234abcde)
+   */
+  public static fromNotificationRuleArn(scope: Construct, id: string, notificationRuleArn: string): INotificationRule {
+    class Import extends Resource implements INotificationRule {
+      readonly notificationRuleArn = notificationRuleArn;
+
+      public addTarget(_target?: INotificationRuleTarget) {
+        return false;
+      }
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * @attribute
+   */
+  readonly notificationRuleArn: string;
+
+  /**
+   * The source config of notification rule
+   */
+  readonly source: NotificationRuleSourceConfig;
```

Comment:
Why is this a field, and especially a public one?

Please turn this into a local variable instead.

Reply:
Don't you want `NotificationRule` to be used alone?
Because the source is required.

Reply:
Already adjusted.